src/iconsManifest/supportedExtensions.ts

```diff
@@ -4360,6 +4360,12 @@ export const extensions: IFileCollection = {
       languages: [languages.sln],
       format: FileFormat.svg,
     },
+    {
+      icon: 'sln',
+      extensions: ['slnx'],
+      languages: [languages.xml],
+      format: FileFormat.svg,
+    },
```

Comment:
This specifies that the XML language should use the `sln` icon.

What I think you intended to do, was update the existing definition for the `sln` icon and add `slnx` to the existing list of extensions. Additionally you should add update the list of extensions for the `sln2` icon below.